diff --git a/lisp/org-clock.el b/lisp/org-clock.el
index cd930e875..449012f84 100644
--- a/lisp/org-clock.el
+++ b/lisp/org-clock.el
@@ -2819,6 +2819,7 @@ a number of clock tables."
             (`month "Monthly report starting on: ")
             (`year "Annual report starting on: ")
             (_ (user-error "Unknown `:step' specification: %S" step))))
+         (step-headers-off (plist-get params :step-headers-off))
          (week-start (or (plist-get params :wstart) 1))
          (month-start (or (plist-get params :mstart) 1))
          (range
@@ -2848,11 +2849,12 @@ a number of clock tables."
             (timestamp (seconds-to-time (org-matcher-time timestamp))))))
     (while (time-less-p start end)
       (unless (bolp) (insert "\n"))
+      (unless step-headers-off
       ;; Insert header before each clock table.
       (insert "\n"
               step-header
               (format-time-string (org-time-stamp-format nil t) start)
-	      "\n")
+	      "\n"))
       ;; Compute NEXT, which is the end of the current clock table,
       ;; according to step.
       (let* ((next
